app/maui-blazor/Services/ServiceImplementations.cs

```diff
@@ -0,0 +1,279 @@
+// for basic app functionality (for now).
+
+public class MauiSessionStorageService : ISessionStorageService
+{
+    private readonly ConcurrentDictionary<string, object> _storage = new(StringComparer.OrdinalIgnoreCase);
+
+    public double Length => _storage.Count;
+
+    public void Clear() => _storage.Clear();
+
+    public TValue? GetItem<TValue>(string key, JsonSerializerOptions? options = null)
+    {
+        // Ignore these...
+        _ = options;
+        
+        return _storage.TryGetValue(key, out var value)
+            ? (TValue?) value
+            : default;
+    }
+
+    public string? Key(double index) => default;
+
+    public void RemoveItem(string key)
+    {
+        _storage.TryRemove(key, out var _);
+    }
+
+    public void SetItem<TValue>(string key, TValue value, JsonSerializerOptions? options = null)
+    {
+        // Ignore these...
+        _ = options;
+        
+        _storage.AddOrUpdate(key, value!, (oldValue, newValue) => newValue);
+    }
+}
+
+public class MauiSpeechSynthesisService : ISpeechSynthesisService
+{
+    private CancellationTokenSource? _cts;
+
+    public bool Paused => throw new NotImplementedException();
+
+    public bool Pending => throw new NotImplementedException();
+
+    public bool Speaking => _cts != null;
+
+    public void Cancel()
+    {
+        _cts?.Cancel();
+        _cts = null;
+    }
+
+    public ValueTask<SpeechSynthesisVoice[]> GetVoicesAsync()
+    {
+        var voice = new SpeechSynthesisVoice
+        {
+            Name = "Default Voice"
+        };
+
+        return ValueTask.FromResult<SpeechSynthesisVoice[]>([voice]);
+    }
+
+    public void Pause()
+    {
+        Cancel();
+
+        // TODO: support pause & resume
+    }
+
+    public void Resume()
+    {
+        // TODO: support pause & resume
+    }
+
+    public async void Speak(SpeechSynthesisUtterance utterance)
+    {
+        _cts = new();
+
+        var current = CultureInfo.CurrentUICulture.Name;
+
+        var locales = await TextToSpeech.Default.GetLocalesAsync();
+        var localeArray = locales.ToArray();
+        var locale = localeArray.FirstOrDefault(l => current == $"{l.Language}-{l.Country}");
+        if (locale is null)
+        {
+            // an exact match was not found, try just the lang
+            var split = current.Split('-');
+            if (split.Length == 1 || split.Length == 2)
+            {
+                // try the first part (or the whole thing if it is just lang)
+                locale = localeArray.FirstOrDefault(l => split[0] == $"{l.Language}");
+            }
+            else
+            {
+                // just go with the first one
+                locale = localeArray.FirstOrDefault();
+            }
+        }
+
+        var options = new SpeechOptions
+        {
+            Locale = locale
+        };
+        await TextToSpeech.Default.SpeakAsync(utterance.Text, options, _cts.Token);
+
+        _cts = null;
```

Comment:
I think this is wrong. If you `null` it out, there's no way to cancel later as you no longer have a reference to the corresponding Token passed into the `SpeakAsync` API above.

Reply:
Fixed.